dartagnan/src/main/java/com/dat3m/dartagnan/wmm/definition/SameScope.java

```diff
@@ -0,0 +1,24 @@
+package com.dat3m.dartagnan.wmm.definition;
+
+import com.dat3m.dartagnan.wmm.Definition;
+import com.dat3m.dartagnan.wmm.Relation;
+
+public class SameScope extends Definition {
+    private final String specificScope;
+
+    public SameScope(Relation r) {
+        super(r);
+        this.specificScope = null;
+    }
```

Comment:
You can call the other constructor `this(r, null)`.

Reply:
Updated